### PR TITLE
Add trigger primitives for Hcal

### LINKED ISSUE #13: Adding in HCal trigger primitives
Starting a thread here to track the developments needed to incorporate a trigger digi producer into the Hcal simulation.

### PR DESCRIPTION
**Pull request by therwig**
This PR is meant as a starting point for the addition of Hcal trigger primitives to the simulation.  Would resolve #13 and requires the updates to ldmx-sw tracked here:
https://github.com/LDMX-Software/ldmx-sw/issues/1013

### CODE REVIEW

File: python/hcal_hardcoded_conditions.py [hunk 1]
```diff
@@ -10,6 +10,16 @@
 
 from LDMX.Conditions.SimpleCSVTableProvider import SimpleCSVIntegerTableProvider, SimpleCSVDoubleTableProvider
 
+# TODO copied from Ecal trigger (ecal_hardcoded_conditions.py)
```
Comment (therwig) [inline python/hcal_hardcoded_conditions.py]: Need to discuss appropriate settings for these parameters w/ @cmantill et al

File: src/Hcal/HcalTrigPrimDigiProducer.cxx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  const ldmx::HgcrocDigiCollection& hcalDigis =
+      event.getObject<ldmx::HgcrocDigiCollection>(digiCollName_, digiPassName_);
+
+  // get the calibration object
+  const conditions::IntegerTableCondition& conditions =
+      getCondition<conditions::IntegerTableCondition>(condObjName_);
+
+  // construct the calculator...
+  ldmx::HgcrocTriggerCalculations calc(conditions);
```
Comment (therwig) [inline src/Hcal/HcalTrigPrimDigiProducer.cxx:34]: Need to discuss  w/ hcal digi experts what updates are required to this calculation.

**Comment by therwig** [comment]
Posting this as a WIP now because while it adds the requisite machinery for the production of Hcal trigger primitives, further discussion is needed on the physical aspects of forming the TPs.

File: src/Hcal/HcalTriggerGeometry.cxx [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+//   // return ldmx::HcalID(triggerCell.layer(), triggerCell.module(),
+//   //                     ptr->second[4].cell());
+//     return ldmx::HcalID();
+// }
+
+ldmx::HcalTriggerID HcalTriggerGeometry::belongsTo(
+    ldmx::HcalID precisionCell) const {
+  unsigned int sstrip=0;
+  unsigned int strip=precisionCell.strip();
+  if (precisionCell.section() == ldmx::HcalID::HcalSection::BACK){
```
Comment (therwig) [inline src/Hcal/HcalTriggerGeometry.cxx]: This adapts the grouping presented by Martin Meier at the SWAN meeting on 19 July 2021.

File: src/Hcal/HcalTrigPrimDigiProducer.cxx (hunk 2 above)
Comment (therwig) [reply src/Hcal/HcalTrigPrimDigiProducer.cxx:34]: done

**Comment by therwig** [comment]
This should be relatively stable now so removing the WIP tag. Thanks @cmantill for the help.

File: src/Hcal/HcalTriggerGeometry.cxx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  for(int iStrip=0;iStrip<4;iStrip++){
+    int strip = iStrip+4*triggerCell.superstrip();
+    if (strip >= hcalGeometry_->getNumStrips(triggerCell.section())) break;
+    retval.push_back( ldmx::HcalDigiID(triggerCell.section(), triggerCell.layer(),
+                                       strip, triggerCell.end()) );
+  }
+  return retval;
+}
+
+  std::vector<ldmx::HcalDigiID> HcalTriggerGeometry::contentsOfSTQ(
```
Comment (cmantill) [inline src/Hcal/HcalTriggerGeometry.cxx:28]: Does this mean that one Super Trigger Quad (STQ) has 4 Quads (16 bars total) divided in 2 layers? i.e. for a given layer you will have only 8 bars that belong to a single STQ - and 8 STQs total?

**Review by cmantill — APPROVED** [review]
@tomeichlersmith I think the PR is ready to merge - just one clarification is needed from my side which is how exactly are STQs formed. Maybe you know or have reference slides? My comment above is just my interpretation from the code.

**Comment by tomeichlersmith** [comment]
I am not sure how the STQs are constructed unfortunately. Last I heard it was actually still up in the air? As in, there isn't an answer yet because we aren't sure what the best strategy would be.

**Pull request merged**